src/main/java/no/sikt/nva/validators/DoiValidator.java

```diff
@@ -156,7 +182,13 @@ private static List<String> extractDoiList(DublinCore dublinCore) {
     }
 
     private static boolean isValidDoiOffline(String doi) {
-        return nva.commons.doi.DoiValidator.validateOffline(doi);
+        return isValidDoi(doi);
+    }
+
+    public static boolean isValidDoi(String doi) {
+        var doiRegex = "^(https?://doi\\.org/)?10.\\d{4,9}/[-._;():A-Z0-9<>\\[\\]]+";
```

Comment:
Where does this magic come from?

Reply:
It is regex that allows doi's of all versions (I think so). It validates correctly invalid doi's and passes "correct" doi's.